packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -413,158 +412,63 @@ class AndroidSdk {
-  // across the  the tool (building Android apps, interacting with the Android SDK, etc.).
-  // Currently, this consistency is fragile since the logic used for building
-  // Android apps exists independently of this method.
-  // See https://github.com/flutter/flutter/issues/124252.
+  /// Callers should instead use the [Java] class.
+  // TODO(andrewkolos): see https://github.com/flutter/flutter/issues/126126.
   static String? findJavaBinary({
+    required Logger logger,
     required AndroidStudio? androidStudio,
     required FileSystem fileSystem,
-    required OperatingSystemUtils operatingSystemUtils,
     required Platform platform,
+    required ProcessManager processManager,
   }) {
-    final String? javaHome = findJavaHome(
+    return Java.find(
+      logger: logger,
       androidStudio: androidStudio,
       fileSystem: fileSystem,
-      operatingSystemUtils: operatingSystemUtils,
       platform: platform,
-    );
-
-    if (javaHome != null) {
-      return fileSystem.path.join(javaHome, 'bin', 'java');
-    }
-
-    // Fallback to PATH based lookup.
-    final String? pathJava = operatingSystemUtils.which(_javaExecutable)?.path;
-    if (pathJava != null) {
-      globals.printTrace('Using java from PATH.');
-    } else {
-      globals.printTrace('Could not find java path.');
-    }
-    return pathJava;
+      processManager: processManager,
+    )?.binary;
   }
 
-  // Returns a user visible String that says the tool failed to parse
-  // the version of java along with the output.
-  static String _formatJavaVersionWarning(String javaVersionRaw) {
-    return 'Could not parse java version from: \n'
-        '$javaVersionRaw \n'
-        'If there is a version please look for an existing bug '
-        'https://github.com/flutter/flutter/issues/'
-        ' and if one does not exist file a new issue.';
-  }
-
-  Map<String, String>? _sdkManagerEnv;
-
   /// Returns an environment with the Java folder added to PATH for use in calling
   /// Java-based Android SDK commands such as sdkmanager and avdmanager.
   Map<String, String> get sdkManagerEnv {
-    if (_sdkManagerEnv == null) {
-      // If we can locate Java, then add it to the path used to run the Android SDK manager.
-      _sdkManagerEnv = <String, String>{};
-      final String? javaBinary = findJavaBinary(
-        androidStudio: globals.androidStudio,
-        fileSystem: globals.fs,
-        operatingSystemUtils: globals.os,
-        platform: globals.platform,
-      );
-      if (javaBinary != null && globals.platform.environment['PATH'] != null) {
-        _sdkManagerEnv!['PATH'] = globals.fs.path.dirname(javaBinary) +
-                                  globals.os.pathVarSeparator +
-                                  globals.platform.environment['PATH']!;
-      }
-    }
-    return _sdkManagerEnv!;
+    // TODO(andrewkolos): Inject Java to avoid referencing the global this deep.
```

Comment:
can you do this in this PR? And if the answer is "no", because it would be too much code, what is our strategy for getting there?

Reply:
Done.